base/server/src/main/java/org/dogtagpki/server/cli/SubsystemIdGeneratorUpdateCLI.java

```diff
@@ -0,0 +1,289 @@
+            socketFactory.setClientCertNickname(authInfo.getClientCertNickname());
+        }
+        socketFactory.init(socketConfig);
+
+        DatabaseConfig dbConfig = cs.getDatabaseConfig();
+
+        if (generatorType.equals("cert")){
+            updateSerialNumberRangeGenerator(
+                    socketFactory,
+                    connInfo,
+                    authInfo,
+                    dbConfig,
+                    baseDN,
+                    generator,
+                    cs.getHostname(),
+                    getSecurePort(cs));
+            cs.commit(false);
+        } else if (generatorType.equals("request")) {
+            updateRequestNumberRangeGenerator(
+                    socketFactory,
+                    connInfo,
+                    authInfo,
+                    dbConfig,
+                    baseDN,
+                    generator);
+            cs.commit(false);
+        } else {
+            throw new EBaseException("Generator type " + generatorType + " not supported.");            
+        }
+    }
+
+    protected void updateSerialNumberRangeGenerator(PKISocketFactory socketFactory, LdapConnInfo connInfo,
+            LdapAuthInfo authInfo, DatabaseConfig dbConfig, String baseDN, IDGenerator newGenerator,
+            String hostName, String securePort) throws Exception {
+        
+        if (newGenerator == IDGenerator.RANDOM && idGenerator != IDGenerator.RANDOM) {
+            logger.debug("Remove serial ranges from configuration");
+            dbConfig.remove(DatabaseConfig.MIN_SERIAL_NUMBER);
+            dbConfig.remove(DatabaseConfig.MAX_SERIAL_NUMBER);
+            dbConfig.remove(DatabaseConfig.SERIAL_INCREMENT);
+            dbConfig.remove(DatabaseConfig.SERIAL_LOW_WATER_MARK);
+            dbConfig.remove(DatabaseConfig.SERIAL_CLONE_TRANSFER_NUMBER);
+            dbConfig.remove(DatabaseConfig.SERIAL_RANGE_DN);
+            return;
+        }
+        if (newGenerator == IDGenerator.LEGACY_2 && idGenerator == IDGenerator.LEGACY) {
+            logger.debug("Repository: Updating ranges entry to hex format");
+
+            LdapBoundConnection conn = new LdapBoundConnection(socketFactory, connInfo, authInfo);
+            try{
+                String rangeDN = dbConfig.getSerialRangeDN() + "," + baseDN;
+
+                String beginSerialNumber = dbConfig.getBeginSerialNumber();
+                dbConfig.setBeginSerialNumber("0x" + beginSerialNumber);
+                
+                String endSerialNumber = dbConfig.getEndSerialNumber();
+                LDAPEntry entrySerial = conn.read("cn=" + beginSerialNumber+"," + rangeDN);
+                LDAPAttribute attrEnd = entrySerial.getAttribute("endRange");
+                if (attrEnd != null) {
+                    endSerialNumber = attrEnd.getStringValues().nextElement();
+                }
+                dbConfig.setEndSerialNumber("0x" + endSerialNumber);
```

Comment:
IIUC the `endRange` attribute is always decimal, but the `dbs.endSerialNumber` for cert is hex, so the `0x` shouldn't be added in all cases.

Reply:
Same problem as above.